src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,223 @@
+import {ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {AztecTypes} from "rollup-encoder/libraries/AztecTypes.sol";
+import {ICurvePool} from "../../interfaces/curve/ICurvePool.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {IConvexBooster} from "../../interfaces/convex/IConvexBooster.sol";
+import {ICurveLpToken} from "../../interfaces/convex/ICurveLpToken.sol";
+import {ICurveRewards} from "../../interfaces/convex/ICurveRewards.sol";
+import {IRepConvexToken} from "../../interfaces/convex/IRepConvexToken.sol";
+
+/**
+ @notice A DefiBridge that allows user to stake Curve LP tokens through Convex Finance and earn boosted CRV 
+ without locking them in for an extended period of time. Plus earning CVX and possibly other rewards. 
+ @notice Staked tokens can be withdrawn (unstaked) any time.
+ @dev Convex Finance mints pool specific Convex LP token, however, not for the staking user (the bridge) directly. 
+ Hence, a storage of interactions with the bridge had to be implemented to know how much Curve LP token 
+ was staked and which Convex LP token was minted to represent this staking action. 
+ Since the Convex LP token is not returned to the bridge, Rollup Processor resp., 
+ a virtual asset is used on output and an interaction `receipt` is created utilizing the virtual asset ID. 
+ The receipt is then used to withdraw the staked means when a virtual asset of the matching ID is provided on input. 
+ @dev Synchronous and stateful bridge that keeps track of deposit interactions with the bridge that are later used for withdrawal.
+ @author Vojtech Kaiser
+ */
+contract ConvexStakingBridge is BridgeBase {
+    using SafeERC20 for IConvexBooster;
+    using SafeERC20 for ICurveLpToken;
+
+    struct PoolInfo {
+        uint96 poolId;
+        address convexToken;
+        address curveRewards;
+    }
+
+    // Convex Finance Booster
+    IConvexBooster public constant BOOSTER = IConvexBooster(0xF403C135812408BFbE8713b5A23a04b3D48AAE31);
+
+    // Pools
+    uint256 public poolLength;
+
+    mapping(address => PoolInfo) public pools;
+
+    // Deployed RepresentingConvexTokens, mapping(Curve LP token => representing Convex token)
+    mapping(address => address) public deployedTokens;
+
+    // Errors
+    error InvalidAssetType();
+    error UnknownAssetA();
+
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {
+        loadPools();
+
+        uint256[] memory criterias = new uint256[](poolLength);
+        uint32[] memory gasUsage = new uint32[](poolLength);
+        uint32[] memory minGasPerMinute = new uint32[](poolLength);
+
+        for (uint256 i = 0; i < poolLength; i++) {
+            (address curveLpToken, , , , , ) = BOOSTER.poolInfo(i);
```

Comment:
Move this to the `loadPool` function since there you are looping over the same data as well. Also you would not have the subsidy criteria set for pools which are added after the deployment of this contract.

Reply:
Resolved